src/Microsoft.OpenApi/Extensions/EnumExtensions.cs

```diff
@@ -42,5 +43,88 @@ public static string GetDisplayName(this Enum enumValue)
+        internal static string GetDisplayName(this ParameterLocation parameterLocation) => parameterLocation switch
+            {
+                ParameterLocation.Query => "query",
+                ParameterLocation.Header => "header",
+                ParameterLocation.Path => "path",
+                ParameterLocation.Cookie => "cookie",
+                _ => parameterLocation.ToString()
+            };
+
+        /// <summary>
+        /// Gets the enum display for name <see cref="ReferenceType" /> without the use of reflection. 
+        /// </summary>
+        /// <param name="referenceType">The enum value.</param>
+        /// <returns>The display string to use.</returns>
+        internal static string GetDisplayName(this ReferenceType referenceType) => referenceType switch
+            {
+                ReferenceType.Schema => "schemas",
+                ReferenceType.Response => "responses",
+                ReferenceType.Parameter => "parameters",
+                ReferenceType.Example => "examples",
+                ReferenceType.RequestBody => "requestBodies",
+                ReferenceType.Header => "headers",
+                ReferenceType.SecurityScheme => "securitySchemes",
+                ReferenceType.Link => "links",
+                ReferenceType.Callback => "callbacks",
+                ReferenceType.Tag => "tags",
+                _ => referenceType.ToString()
+            };
+
+        /// <summary>
+        /// Gets the enum display for name <see cref="OperationType" /> without the use of reflection. 
+        /// </summary>
+        /// <param name="operationType">The enum value.</param>
+        /// <returns>The display string to use.</returns>
+        internal static string GetDisplayName(this OperationType operationType) => operationType switch
+            {
+                OperationType.Get => "get",
+                OperationType.Put => "put",
+                OperationType.Post => "post",
+                OperationType.Delete => "delete",
+                OperationType.Options => "options",
+                OperationType.Head => "head",
+                OperationType.Patch => "patch",
+                OperationType.Trace => "trace",
+                _ => operationType.ToString()
+            };
+
+        /// <summary>
+        /// Gets the enum display for name <see cref="SecuritySchemeType" /> without the use of reflection. 
+        /// </summary>
+        /// <param name="securitySchemeType">The enum value.</param>
+        /// <returns>The display string to use.</returns>
+        internal static string GetDisplayName(this SecuritySchemeType securitySchemeType) => securitySchemeType switch
+            {
+                SecuritySchemeType.ApiKey => "apiKey",
+                SecuritySchemeType.Http => "http",
+                SecuritySchemeType.OAuth2 => "oauth2",
+                SecuritySchemeType.OpenIdConnect => "openIdConnect",
+                _ => securitySchemeType.ToString()
```

Comment:
don't you need equivalent parsing methods for the parsing scenarios?

Reply:
I took a look at the parsing logic and it looks like the only place we use `GetDisplayName` in the reader implementation is in `OpenApiParameterDeserializer`? With that in mind, I've make the `GetDisplayName` overload for `ParameterLocation` public to support that particular scenario.